plugins/vbe/fu-plugin-vbe.c

```diff
@@ -0,0 +1,279 @@
+	}
+	method_name = p + 1;
+
+	/* find this update mechanism */
+	for (driver = driver_list; driver->name; driver++) {
+		if (!strcmp(method_name, driver->name)) {
+			meth = g_malloc(sizeof(struct FuVbeMethod));
+			meth->vbe_method = p + 1;
+			meth->node = node;
+			meth->driver = driver;
+			g_info("Update mechanism: %s", meth->vbe_method);
+			*methp = meth;
+			return TRUE;
+		}
+	}
+	g_error("No driver for VBE method '%s'", method_name);
+
+	return FALSE;
+}
+
+static gboolean process_system(FuPluginData *priv, gchar *fdt, gsize fdt_len,
+			       GError **error)
+{
+	int ret, parent, node;
+	int found;
+
+	ret = fdt_check_header(fdt);
+	if (ret) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
+			    "System DT is corrupt (%s)",
+			    fdt_strerror(ret));
+		return FALSE;
+	}
+	if (fdt_totalsize(fdt) != fdt_len) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
+			    "System DT size mismatch (header=%x, file=%zx)",
+			    fdt_totalsize(fdt), fdt_len);
+		return FALSE;
+	}
+	parent = fdt_path_offset(fdt, NODE_PATH);
+	if (parent < 0) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
+			    "Missing node '%s' (%s)", NODE_PATH,
+			    fdt_strerror(ret));
+		return FALSE;
+	}
+
+	/* Create a device for each subnode */
+	found = 0;
+	for (node = fdt_first_subnode(fdt, parent); node > 0;
+	     node = fdt_next_subnode(fdt, node)) {
+		struct FuVbeMethod *meth;
+
+		if (vbe_locate_device(fdt, node, &meth)) {
+			found++;
+		} else {
+			g_warning("Cannot locate device for node '%s'",
+				  fdt_get_name(fdt, node, NULL));
+		}
+		priv->methods = g_list_append(priv->methods, meth);
+	}
+
+	if (found) {
+		g_info("VBE update methods: %d", found);
+	} else {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+			    "No valid VBE update mechanism found");
+	}
+
+
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_vbe_startup(FuPlugin *plugin, FuProgress *progress, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_autofree gchar *vbe_dir = NULL;
+	g_autofree gchar *state_dir = NULL;
+	gchar *buf, *bfname;
+	gsize len;
+	int ret;
+
+	ret = fit_test();
+	if (ret) {
+		g_info("fit_test failed: %d", ret);
+		return FALSE;
+	}
+
+	/* Get the VBE directory */
+	state_dir = fu_common_get_path(FU_PATH_KIND_LOCALSTATEDIR_PKG);
+	vbe_dir = g_build_filename(state_dir, "vbe", NULL);
+	priv->vbe_dir = g_steal_pointer(&vbe_dir);
+
+	bfname = g_build_filename(priv->vbe_dir, SYSTEM_DT, NULL);
+	if (!g_file_get_contents(bfname, &buf, &len, error)) {
+		/* Check if we have a kernel device tree */
+		g_warning("Cannot find system DT '%s'", bfname);
+
+		/* Read in the system info */
+		g_free(bfname);
+		bfname = g_build_filename(KERNEL_DT, NULL);
+		if (!g_file_get_contents(bfname, &buf, &len, error)) {
+			g_warning("No kernel device tree '%s'", bfname);
+			g_free(bfname);
+			return FALSE;
+		}
+	}
+	g_log(G_LOG_DOMAIN, G_LOG_LEVEL_INFO, "Processing system DT '%s'",
+	      bfname);
```

Comment:
just use `g_debug`

Reply:
Then how do I make the messages appear?

I tried:

G_MESSAGES_DEBUG=Vbe sudo -E build/src/fwupdtool install ...

but see nothing

Reply:
thank you